snapcraft/tests/plugins/python/test_pip.py

```diff
@@ -0,0 +1,412 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+
+import fixtures
+from unittest import mock
+
+from testtools.matchers import (
+    Contains,
+    Equals,
+)
+
+from snapcraft.plugins._python import (
+    _pip,
+    errors,
+)
+
+from snapcraft import (
+    tests
+)
+
+
+def _create_python_binary(base_dir):
+    python_command_path = os.path.join(
+        base_dir, 'usr', 'bin', 'pythontest')
+    os.makedirs(os.path.dirname(python_command_path))
+    open(python_command_path, 'w').close()
+    return python_command_path
+
+
+class PipRunTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        patcher = mock.patch('snapcraft.internal.common.run_output')
+        self.mock_run_output = patcher.start()
+        self.addCleanup(patcher.stop)
```

Comment:
Your tests are very nice, but I still prefer my implementation with a fake pip fixture.
As I said during our discussion, I'm not sure what's best, so I'll let Sergio break the tie.

Reply:
After our discussion today, I think I agree that the fake fixture is best. However, I'd rather have that as a follow-up.